crates/solidity/src/solc/standard_json/input/settings/selection/file/flag.rs

```diff
@@ -57,6 +60,7 @@ impl std::fmt::Display for Flag {
             Self::EVMBC => write!(f, "evm.bytecode"),
             Self::EVMDBC => write!(f, "evm.deployedBytecode"),
             Self::Assembly => write!(f, "evm.assembly"),
+            Self::Ir => write!(f, "ir"),
```

Comment:
we have evm.bytecode, so why it is not evm.ir?

Reply:
`solc` puts EVM related artifacts into the `evm` objects ([see the docs for where is what](https://docs.soliditylang.org/en/latest/using-the-compiler.html#output-description)).

So the IR is probably not there because it isn't directly tied to EVM. Even` though it could because supposedly there are multiple YUL dialects, but this never manifested and there is just the EVM YUL dialect. If there would be a second one it's unclear to me where they would put it because the root object already implicitly contains the YUL _EVM_ IR dialect. But we don't need to worry about it since if it ever happens we just adapt to whatever they are doing.

Reply:
ir is just an unoptimized YUL output that might be used down the line in `foundry` itself and it seemed like a harmless thing to add support for.

Reply:
`revive` does process the optimized IR. So why does foundry need it?

Reply:
`Ir` is used for forge inspect and that's basically it.